### ISSUE TITLE
Analog of XCTestObservation

### ISSUE THREAD

**Issue by ypopovych** [apple/swift-testing#469]
### Description

It will be good to have a way to observe testing process. Analog to the `XCTestObservation` and `XCTestObservationCenter` in the XCTest

**Comment by ypopovych** [comment]
Seems as library has `Event` API already.  API to register event handlers should be added (maybe as dynamic library through command-line arguments) or through some macros in code.

**Comment by stmontgomery** [comment]
I agree this is something we want to expose as public API eventually.

Tracked internally as rdar://129609023

**Comment by grynspan** [comment]
We have a lot of the moving parts already. What we don't have is a way to hook into the event stream arbitrarily.

Currently, there is support for streaming JSON descriptions of events to an arbitrary file when using `swift test`. Xcode doesn't offer a way to enable that functionality, but it _could_ be done.

Would that be sufficient for your use case?

**Comment by grynspan** [comment]
Experimental implementation on branch jgrynspan/469-observer-attribute.

**Comment by ypopovych** [comment]
Hello.

Do you have some API implemented for that purpose in the latest versions?

I tried to find out a way to do that, but seems as it's impossible to add custom event handler, at least in case of Xcode (like override entry point).
